### PR TITLE
feat(security-actions/scan-docker-image): support `trivy_db_cache` as alternate

### PR DESCRIPTION
**Pull request by dndx**
input which allows caching the Trivy DB to get around rate limiting issues

Trivy (ab)uses GHCR OCI as it's config update mechanism, and this is being heavily (targeted) rate limited by GitHub.

Inspired by https://github.com/aquasecurity/trivy/discussions/7591, this PR adds two optional arguments to the `security-actions/scan-docker-image` image that allows us to skip fetching fro GHCR completely:

```
trivy_db_cache:
    description: 'GitHub repository containing Trivy DB cache (format: owner/repo@ref). Database should be named `db.tar.gz` on the default branch.'
    required: false
trivy_db_cache_token:
    description: 'Token for accessing `trivy_db_cache`.'
    required: false
```

When provided, we will checkout the specific repo's branch using the PAT, look for a `db.tar.gz` file from it and untar it to the filesystem. We then instruct Trivy to not update it's database and instead use the local database only.

This ensures the action will always succeed because a database will always be available. https://github.com/Kong/trivy-db-mirror contains a [workflow](https://github.com/Kong/trivy-db-mirror/blob/master/.github/workflows/update-database.yml) that automatically updates it's database every 6 hours, but update can also be manually triggered via [Run workflow button on this page](https://github.com/Kong/trivy-db-mirror/actions/workflows/update-database.yml).

This PR also removed running Trivy inside the Docker container, as it appears that GitHub Actions does not support volume mounting with the `docker://` method. Instead we simply download the `trivy` executable from the official release link and throw it into `/usr/local/bin`.

According to [my tests](https://github.com/Kong/kong-ee/actions/runs/11683216638/job/32534201531?pr=10626), the scan image workflow now always succeeds.

Sister PR on Kong-ee: https://github.com/Kong/kong-ee/pull/10626

### CODE REVIEW

**Comment by saisatishkarra** [comment]
Based on the below discussions for the mentioned Issue, 
-  **Issue**: https://github.com/aquasecurity/trivy/issues/7938
- **Upstream related Discussions:** 
   - https://github.com/aquasecurity/trivy/discussions/7668
   - https://github.com/aquasecurity/trivy/discussions/7699
   - https://github.com/aquasecurity/trivy/discussions/7951 
- **Conclusion:** To **avoid trivy rate-limiting issues** and **re-enable docker cis scan**
  - By default, the action leverage the the updated upstream `mirror.gcr.io` for up-to-date CIS DB checks 
  - Anyone requiring availability at all times, explicitly set `trivy_db_cache: Kong/trivy-db-mirror@master` and `trivy_db_cache_token` to clone DB mirror repository accessible internally within `Kong` org

**Review by saisatishkarra — APPROVED** [review]
LGTM

File: security-actions/scan-docker-image/action.yml [hunk 1]
```diff
@@ -302,13 +308,60 @@ runs:
...
+        curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/main/contrib/install.sh | sh -s -- -b /usr/local/bin v0.57.1
+
+    - name: Checkout Trivy DB cache
+      if: inputs.trivy_db_cache != ''
+      uses: actions/checkout@v4
+      with:
+        repository: ${{ steps.parse_cache.outputs.repository }}
+        ref: ${{ steps.parse_cache.outputs.ref }}
+        path: trivy-db-cache
+        token: ${{ inputs.trivy_db_cache_token }}
```
Comment (saisatishkarra) [inline security-actions/scan-docker-image/action.yml:336]: @dndx  when a PAT token is specified, does that token user need to be explicitly allowed access on the `trivy-db-cache` repo? If that's the case, we need to document where such GH repo requests can be requested as this adds additional steps. Not a blocker but adds explicit step to invite user to the mirroring repo.

If only the input PAT and `read` permission are sufficient, this flow looks good to me.

Comment (saisatishkarra) [reply security-actions/scan-docker-image/action.yml:336]: An Attempt to use the Org wide GH APP instead  [isn't working due to lack of vars context in composite actions](https://github.com/orgs/community/discussions/49689) with a ([failed run](https://github.com/Kong/public-shared-actions/actions/runs/12052513771)) and the current App ID is only available to internal and private repositories.

Comment (saisatishkarra) [reply security-actions/scan-docker-image/action.yml:336]: The instructions to use cached trivy db are updated at: ttps://github.com/Kong/public-shared-actions/blob/c0bb00e19989975898ff19be4f5cc5bae356c10c/security-actions/scan-docker-image/README.md#user-provided-input-parameters

Comment (saisatishkarra) [reply security-actions/scan-docker-image/action.yml:336]: fyi: @vsofronievk  for review / suggestions. Otherwise resolve and review for approval

**Review by pankajmouriyakong — APPROVED** [review]
LGTM

**Pull request merged**